navigator.js

```diff
@@ -49,17 +49,17 @@ var grab = false;
 var ActionDispatcher = class {
   constructor() {
     this.actor = new Clutter.Actor();
+    this.actor.set_flags(Clutter.ActorFlags.REACTIVE);
     Main.uiGroup.add_actor(this.actor);
-    if (!Main.pushModal(this.actor)) {
-      // Probably someone else has a pointer grab, try again with keyboard only
-      if (
-        !Main.pushModal(this.actor, {
-          options: Meta.ModalOptions.POINTER_ALREADY_GRABBED,
-        })
-      )
-        throw new Error("Could not grap a modal");
+
+    let grabHandle = Main.pushModal(this.actor);
+    // We expect at least a keyboard grab here
+    if ((grabHandle.get_seat_state() & Clutter.GrabState.KEYBOARD) === 0) {
+      Main.popModal(grabHandle);
+      log("Failed to grab modal");
+      throw new Error("Could not grab modal");
```

Comment:
@Thesola10 you were returning false here instead of throwing an error, do you remember if there was a reason for that?

Reply:
Actually maybe this was from @xanf originally, but same question: Is there a reason you stopped throwing an error here?